src/dataServices/dynamoDbDataService.ts

```diff
@@ -385,6 +385,25 @@ export class DynamoDbDataService implements Persistence, BulkDataAccess {
         return exportJob;
     }
 
+    async getActiveSubscriptions(params: { tenantId?: string }): Promise<Record<string, any>[]> {
+        this.assertValidTenancyMode(params.tenantId);
+        const subscriptionQuery = DynamoDbParamBuilder.buildGetActiveSubscriptions(params.tenantId);
+        let queryResponse = await this.dynamoDb.query(subscriptionQuery).promise();
+        const subscriptions: Record<string, any>[] = [];
+        do {
+            queryResponse.Items?.forEach((response) => {
+                const item = DynamoDBConverter.unmarshall(response);
+                subscriptions.push({ [item.id]: item });
+            });
+            if (queryResponse.LastEvaluatedKey) {
+                subscriptionQuery.ExclusiveStartKey = queryResponse.LastEvaluatedKey;
+                // eslint-disable-next-line no-await-in-loop
+                queryResponse = await this.dynamoDb.query(subscriptionQuery).promise();
+            }
+        } while (queryResponse.LastEvaluatedKey);
```

Comment:
refactor so that `let queryResponse = await this.dynamoDb.query(subscriptionQuery).promise();` happens only once at the beginning of the do while. It's clearer that way

Reply:
Refactored!